include/SFML/Graphics/Glsl.inl

```diff
@@ -81,7 +81,7 @@ struct Matrix
         copyMatrix(transform, *this);
     }
 
-    float array[Columns * Rows]; //!< Array holding matrix data
+    std::array<float, Columns * Rows> array; //!< Array holding matrix data
```

Comment:
I'm not sure if a 2D array is needed or brings us any advantage here. It's quite common in gamedev to represent matrices as 1D arrays.

Reply:
It's not necessary but it better expresses our intent. While we're currently using a 1D array, what we really mean to express is how there are two independent dimensions here. Matrices are inherently 2 dimensional. Using a 2D array would let us much more clearly write code that needs to access certain elements by their row and column index. It's not immediately obvious what index 13 means since you have to know the size of the matrix and do some mental math, but if you express that as row 3 column 2 then you can much more easily understand what's going on plus that code is much more resilient to potential changes in the size of the array.

Reply:
I am also not sure it is worth doing. I agree it is semantically more correct to implement a matrix as a 2D array. However, a span over a 2D array would still be 1D (unless we add a multi-dimensional span class) so we would still have to use 1D indices at some places.

Reply:
There is no obvious consensus and this can be kept for later so I resolve this discussion for now.